### PR TITLE
network: only add --quiet-* options to dnsmasq if version supports them

### LINKED ISSUE #3757: lxd 2.17 init issues with dnsmasq
The template below is mostly useful for bug reports and support questions.
Feel free to remove anything which doesn't apply to you and add more information where it makes sense.

# Required information

 * Distribution: CentOS
 * Distribution version: 7
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.12.10-1.el7.elrepo.x86_64
   * LXC version: 2.17
   * LXD version: 2.17
   * Storage backend in use: Default

# Issue description
After looking at my other VM, which is running 2.16 and this one, which is 2.17, I realized that the 
difference between the two is the addition of: "--quiet-dhcp --quiet-dhcp6 --quiet-ra". Once I removed those and pasted to the command line, dnsmasq no longer had an issue with it.

# Steps to reproduce
_lxd init_

Do you want to configure a new storage pool (yes/no) [default=yes]? 
Name of the new storage pool [default=default]: 
Name of the storage backend to use (dir, btrfs, lvm) [default=dir]: 
Would you like LXD to be available over the network (yes/no) [default=no]? 
Would you like stale cached images to be updated automatically (yes/no) [default=yes]? 
Would you like to create a new network bridge (yes/no) [default=yes]? 
What should the new bridge be called [default=lxdbr0]? 
What IPv4 address should be used (CIDR subnet notation, “auto” or “none”) [default=auto]? 
What IPv6 address should be used (CIDR subnet notation, “auto” or “none”) [default=auto]? 
error: Failed to run: dnsmasq --strict-order --bind-interfaces --pid-file=/var/lib/lxd/networks/lxdbr0/dnsmasq.pid --except-interface=lo --interface=lxdbr0 --quiet-dhcp --quiet-dhcp6 --quiet-ra --listen-address=10.195.61.1 --dhcp-no-override --dhcp-authoritative --dhcp-leasefile=/var/lib/lxd/networks/lxdbr0/dnsmasq.leases --dhcp-hostsfile=/var/lib/lxd/networks/lxdbr0/dnsmasq.hosts --dhcp-range 10.195.61.2,10.195.61.254,1h --listen-address=fd42:eb7:2b20:965::1 --enable-ra --dhcp-range ::,constructor:lxdbr0,ra-stateless,ra-names -s lxd -S /lxd/ --conf-file=/var/lib/lxd/networks/lxdbr0/dnsmasq.raw -u lxd: dnsmasq: bad command line options: try --help

### PR DESCRIPTION
**Pull request by albertodonato**
Fixes #3757

### CODE REVIEW

**Comment by albertodonato** [comment]
jenkins: test this please

File: lxd/networks.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/blang/semver"
```
Comment (stgraber) [inline lxd/networks.go]: Can't do that, it's a new package which hasn't gone through a security review.

We may be able to make our version checking code from LVM generic to cover this case.

**Comment by albertodonato** [comment]
@stgraber I dropped the use of the external library

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
This breaks LVM

**Comment by stgraber** [comment]
Sent the first two commits as a separate branch that we can get in 2.18.
Moving LVM over to this parser is fine, but we need to make sure that it does parse those versions properly (which it doesn't appear to now).

**Comment by albertodonato** [comment]
@stgraber should be fixed now

**Pull request merged**